### ISSUE TITLE
 feature: set priority per tracker (for all torrents, including new)

### ISSUE THREAD

**Issue by ilf** [transmission/transmission#6425]
### What is the issue?

Currently, users can set priority per torrent.
It can be desirable to set priority per tracker - for all torrents, including new ones.

Originally [filed with transmission-remote-gtk](https://github.com/transmission-remote-gtk/transmission-remote-gtk/issues/243), but the maintainer sais [this should be in daemon](https://github.com/transmission-remote-gtk/transmission-remote-gtk/issues/243#issuecomment-1867760343).

### Which application of Transmission?

transmission-daemon

**Comment by killemov** [comment]
> It can be desirable to set priority per tracker - for all torrents, including new ones.

So ... filter torrents by tracker and then set the priority for the filtered torrents? (As in: Is this a one shot action, not actively tied to the tracker, or is it?)

What is YOUR actual use-case for this?

**Comment by ilf** [comment]
@killemov: Yes, that's what I currently do. But this works only for existing torrents, and I have to do it manually. I would love to also have this for new torrents - and I want software to do this instead of humans. :)

**Comment by killemov** [comment]
I get that you have to do it manually, but why do it at all? Bandwidth priority isn't THAT important. So again, what is the use-case?

**Comment by ilf** [comment]
@killemov: I'm confused by your comment. It's okay if you have no use for torrent priority. But clearly I do, so do others, that's why it's in transmission (and any other torrent clients).

**Comment by killemov** [comment]
AFAICT the priority setting of a torrent only has a marginal influence of it's actual download speed. Not that high = 10x normal = 10x low or something of that order. And I'm honestly interested in why you would couple torrent priority with tracker(s). And that is why asked for the use-case. You clearly have use for it, so what is it?

**Comment by ilf** [comment]
@killemov: First, priority is about more than "download speed", it's also about upload bandwidth and number of peer connections - both of which can be much more limited than download bandwidth.

Second, the priority setting does have an effect. If it doesn't, we should remove the feature entirely.

And yes, I regularly adjust the priority to low or high for all torrents of particular trackers.

For example: I want to set http://bttracker.debian.org as priority high, since it's run by a non-profit community project. And I want to set https://torrent.ubuntu.com as priority low, since it's run by a for-profit company.

People using private trackers may have other reasons, like prioritizing either upload-data or seed-time.

**Comment by killemov** [comment]
So it appears you wish to set the bandwidth priority for seeding (only?). Okay, but I'm not really sure about the impact of this setting for upload speeds. 

> Second, the priority setting does have an effect. If it doesn't, we should remove the feature entirely.

You have tested this? Also, is it possible for you to have conflicting trackers within a single torrent?

This could be one of those "power user features" @ckerr might be interested in, but I wouldn't hold my breath.
